pages/api/mechanic/updateAvailability.ts

```diff
@@ -10,19 +10,17 @@ export default async function handler(
 ) {
   if (req.method === "POST") {
     try {
-      // Call the authorize middleware to extract user ID from token
       authorize(req, res, async () => {
         const user = req["user"];
-        const userId = user.id;
 
         // Validate user ID
-        if (!userId) {
+        if (!user?.id) {
```

Comment:
Is it possible to not have the userId in the query? 


I observed we do this check only in specific files, so does this mean that this specific api call query may or may not have the userId in it? 

Does all the other api call's query has the userId where we did't do this check?